### PR TITLE
Add test exercising lldp configuration with nmpolicy

### PR DESCRIPTION
**Pull request by rhrazdil**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind enhancement

**What this PR does / why we need it**:
This PR bumps nmpolicy dependency to allow configuration of boolean values.
In the second commit, this feature is used in an e2e test that configures lldp with nmpolicy capture.
In the third commit, an example of enabling lldp on all ethernet interfaces that are up is added.

**Special notes for your reviewer**:

**Release note**:

```release-note
bump nmpolicy to v0.5.0
```

### CODE REVIEW

File: test/e2e/handler/lldp_with_nmpolicy_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		}
+	}
+
+	interfacesWithLldpEnabledState := nmstate.NewState(`interfaces: "{{ capture.ethernets-lldp.interfaces }}"`)
+
+	BeforeEach(func() {
+		By("Enabling LLDP on up ethernet interfaces")
+		setDesiredStateWithPolicyAndCapture(lldpEnabledPolicyName, interfacesWithLldpEnabledState, configureLldpOnEthernetsCapture("true"))
+
+		DeferCleanup(func() {
```
Comment (qinqon) [inline test/e2e/handler/lldp_with_nmpolicy_test.go:48]: We can put all the AfterEach on the DeferCleanup right ? also look slike we are repeting the DeferCleanup at the AfterEach.

File: test/e2e/handler/lldp_with_nmpolicy_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			deletePolicy(lldpDisabledPolicyName)
+		})
+
+		By("Wait for policy to be ready")
+		policy.WaitForAvailablePolicy(lldpDisabledPolicyName)
+	})
+
+	It("should enable LLDP on all ethernet interfaces that are up", func() {
+		By("Wait for policy to be ready")
+		policy.WaitForAvailablePolicy(lldpEnabledPolicyName)
```
Comment (qinqon) [inline test/e2e/handler/lldp_with_nmpolicy_test.go:66]: Let's move this to BeforeEach, the It just check the values.

**Review by qinqon — CHANGES_REQUESTED** [review]
Can we add an example of the LLDP ?

File: test/e2e/handler/lldp_with_nmpolicy_test.go (hunk 1 above)
Comment (rhrazdil) [reply test/e2e/handler/lldp_with_nmpolicy_test.go:49]: Added an example yaml, moved AfterEach into DeferCleanup

File: test/e2e/handler/lldp_with_nmpolicy_test.go (hunk 2 above)
Comment (rhrazdil) [reply test/e2e/handler/lldp_with_nmpolicy_test.go]: Done

File: docs/examples/lldp.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: nmstate.io/v1
```
Comment (qinqon) [inline docs/examples/lldp.yaml:1]: we should test this example

Comment (rhrazdil) [reply docs/examples/lldp.yaml:1]: we can do so once we have a release with the lldp.yaml, otherwise it break upgrade

Comment (qinqon) [reply docs/examples/lldp.yaml:1]: why we cannot do it on this PR ? there is  example test suite at test/e2e

**Review by qinqon — APPROVED** [review]
/lgtm
/approve

**Pull request merged**